README.md

```diff
@@ -888,3 +813,444 @@ Optional.Some(3).tap(x  =>  console.log("Hello") ||  x  +  1);
 Optional.None().tap(x  =>  console.log("Test"));
 
 ```
+
+### `Result`
+
+A model for the result of a computation chain.
+
+#### Ok
+
+`Result` subtype for truthy values.
+
+#### Error
+
+`Result` subtype for errors and falsy values.
+
+#### Aborted
+
+Like `Error`, but unrecoverable.
+
+#### Pending
+
+`Result` subtype for asynchronous data-types.
+
+#### `Result` construction
+
+A `Result` can be constructed by directly calling the subtype that you'd like to use:
+
+```javascript
+const result1 = Result.Ok(3);
+const result2 = Result.Error({ message: 'Ooops' });
+const result3 = Result.Pending();
+const result4 = Result.Aborted();
+```
+
+More often, you'll find yourself using either one of `when`, `try` or `expect`.
+
+#### `Result`'s `when(test, value, error)`
+
+`when` evaluates the first argument passed to it, and returns `value` wrapped in an `Ok` if it's truthy, else it returns `error` wrapped in an `Error`:
+
+```javascript
+// This evaluates to Ok(3)
+Result.when(true, 3, 'Oops');
+
+// This evaluates to Error('Oops')
+Result.when(false, 3, 'Oops');
+```
+
+#### `Result`'s `try(λ)`
+
+Takes a `λ` and returns the result wrapped in a `Result`.
+
+```javascript
+// Evaluates to Ok(3)
+Result.try(() => Result.Ok(3));
+
+// Also evaluates to Ok(3)
+Result.try(() => 3);
+
+// Evaluates to Error()
+Result.try(() => null);
+```
+
+#### `Result`'s `expect(object)`
+
+Takes any type of object and return it wrapped in a `Result` with the appropriate subtype.
+
+```javascript
+// This evaluates to Result.Error
+Result.expect(null);
+
+// This evaluates to Result.Ok()
+Result.expect(3);
+
+// This evaluates to either a Result.Ok or a Result.Aborted
+Result.expect(axios.get('/api/version'));
```

Comment:
What's the logic? I suppose this is critical information (not sure, but I think it has to do with failing promises?)